### PR TITLE
feat: BECS-72 implement the guest user can register to become a customer in system

### PR DESCRIPTION
**Pull request by phucnguyen97**
Ticket: https://quang-tran.atlassian.net/browse/BECS-72

### CODE REVIEW

File: src/apps/account/models/user.ts [hunk 1]
```diff
@@ -8,6 +8,12 @@ export class User extends BaseModel<User> {
   @prop({ required: true, limit: 100 })
   public username: string;
 
+  @prop({ required: true, limit: 50 })
+  public password: string;
+
+  @prop({ required: true, limit: 50 })
+  public rePassword: string;
```
Comment (quangtran85) [inline src/apps/account/models/user.ts]: I think we don't need to store this value, plz remove it.

File: src/apps/account/services/user.service.ts [hunk 2]
```diff
@@ -91,6 +108,7 @@ export class UserService {
       throw new AppException(errorCode, message, httpCode);
     }
 
+    data.password = await bcrypt.hash(data.password, 10);
```
Comment (quangtran85) [inline src/apps/account/services/user.service.ts]: The password would be hashed from the auth service, don't need to hash password here. Pls remove this.

File: src/apps/account/models/user.ts [hunk 3]
```diff
@@ -17,11 +23,33 @@ export class User extends BaseModel<User> {
...
+  @prop({ required: true, limit: 50 })
+  public zipCode: string;
+
+  @prop({ default: false })
+  public isMember: boolean;
+
   toEntity(user = new User()): User {
     user = this.toBaseEntity(user);
     user.firstName = this.firstName;
     user.lastName = this.lastName;
```
Comment (quangtran85) [inline src/apps/account/models/user.ts]: firstName & lastName are not mentioned in the specs, therefore pls remove them out.

**Review by quangtran85 — COMMENTED** [review]
@phucnguyen97 I have some comments, pls check them.

**Comment by phucnguyen97** [comment]
> @phucnguyen97 I have some comments, pls check them.

@quangtran85 I have fixed, please help me review

**Review by quangtran85 — APPROVED** [review]
LGTM!

**Pull request merged**